projects/canopy/src/lib/button/button.stories.ts

```diff
@@ -74,110 +58,306 @@ class ButtonStoryComponent {
+      imports: [LgButtonModule, LgIconModule],
+    }),
+  ],
   parameters: {
-    decorators: [
-      withKnobs,
-      moduleMetadata({
-        declarations: [ButtonStoryComponent],
-        imports: [LgButtonModule, LgIconModule],
-      }),
-    ],
-    notes: {
-      markdown: notes,
+    docs: {
+      description: {
+        component: notes,
+      },
+    },
+  },
+  argTypes: {
+    variant: {
+      options: [...buttonVariants],
+      defaultValue: 'solid-primary',
+      table: {
+        type: {
+          summary: 'ButtonVariant',
+        },
+      },
+      control: {
+        type: 'select',
+      },
+    },
+    content: {
+      description: 'Text content within the button',
+    },
+    icon: {
+      description: 'Icon to display',
+      options: ['None', ...iconsArray.map((i) => i.name)],
+      table: {
+        type: {
+          type: { summary: 'string' },
+        },
+      },
+      control: {
+        type: 'select',
+      },
+    },
+    _variant: {
+      table: { disable: true },
+    },
+    class: {
+      table: { disable: true },
+    },
+    hostElement: {
+      table: { disable: true },
     },
   },
 };
 
-interface KnobsConfig {
-  disabled?: boolean;
-  fullWidth?: boolean;
-  iconButton?: boolean;
-  loading?: boolean;
-  showIcon?: boolean;
-  size?: ButtonSize;
-  variant: ButtonVariant;
-}
+const iconArgType = {
+  description: 'Icon to display',
+  options: iconsArray.map((i) => i.name),
+  defaultValue: iconsArray[0].name,
+  table: {
+    type: {
+      type: { summary: 'string' },
+    },
+  },
+  control: {
+    type: 'select',
+  },
+};
 
-const createBtnStory = (config: KnobsConfig) => ({
-  template: `
-    <lg-button-story
-      [disabled]="disabled"
-      [fullWidth]="fullWidth ? true : null"
-      [icon]="icon"
-      [iconButton]="iconButton"
-      [iconPosition]="iconPosition"
-      [loading]="loading ? true : null"
-      [showIcon]="showIcon"
-      [size]="size"
-      [variant]="variant"
-      [content]="content"
-      >
-  </lg-button-story>
-  `,
-  props: {
-    disabled: boolean('disabled', false, propsGroupId),
-    content: text('content', 'Button', contentGroupId),
-    fullWidth: boolean('fullWidth', config.fullWidth, propsGroupId),
-    icon: select(
-      'icon',
-      iconsArray.map((icon) => icon.name),
-      'add',
-      contentGroupId,
-    ),
-    iconButton: boolean('iconButton', config.iconButton, propsGroupId),
-    iconPosition: select('iconPosition', ['left', 'right'], 'right', propsGroupId),
-    loading: boolean('loading', config.loading, propsGroupId),
-    showIcon: boolean('show icon', config.showIcon, contentGroupId),
-    size: select('size', ['sm', 'md'], 'md', propsGroupId),
-    variant: select('variant', buttonVariants, config.variant, propsGroupId),
+const defaultArgValues = {
+  content: 'Click me',
+  disabled: false,
+  fullWidth: false,
+  iconButton: false,
+  loading: false,
+  icon: 'None',
```

Comment:
My personal preference would be to use `null` instead of this string

Reply:
As discussed we will keep `None`

Reply:
As discussed on a call, we'll leave this in as it looks better in the controls drop down 👍